errors/errors.js

```diff
@@ -0,0 +1,91 @@
+const JSONAPIError = require('jsonapi-serializer').Error;
+const _ = require('lodash');
+
+/**
+ * @summary Construct error object
+ * @function
+ * @param {string} status The HTTP status code
+ * @param {string} title A short, human-readable summary of the problem
+ * @param {string} code An application-specific error code
+ * @param {string} detail A human-readable explanation
+ * @returns {Object} An error object
+ */
+const error = (status, title, code, detail) => ({
+  status,
+  title,
+  code,
+  detail,
+  links: { about: `https://developer.oregonstate.edu/documentation/error-reference#${code}` },
+});
+
+/**
+ * @summary [400] Return a Bad Request error object
+ * @function
+ * @param {[string]} details An array of bad request details
+ * @returns {Object} Bad Rquest error object
+ */
+const badRequest = (details) => {
+  const badRequests = [];
+  _.forEach(details, detail => badRequests.push(error('400', 'Bad Request', '1400', detail)));
+  return new JSONAPIError(badRequests);
+};
+
+/**
+ * @summary [401] Return a Unauthorized error object
+ * @function
+ * @returns {Object} Unauthorized error object
+ */
+const unauthorized = () => new JSONAPIError(error('401', 'Unauthorized', 'Unauthorized'));
+
+/**
+ * @summary [403] Return a Forbidden error object
+ * @function
+ * @param {string} detail
+ * @returns {Object} Unauthorized error object
+ */
+const forbidden = detail => new JSONAPIError(error('403', 'Forbidden', detail));
+
+/**
+ * @summary [404] Return a Not Found error object
+ * @function
+ * @param {string} detail
+ * @returns {Object} Not Found error object
+ */
+const notFound = detail => new JSONAPIError(error('404', 'Not found', detail));
+
+/**
+ * @summary [409] Return a Conflict error object
+ * @function
+ * @param {string} detail
+ * @returns {Object} Conflict error object
+ */
+const conflict = detail => new JSONAPIError(error('409', 'Conflict', detail));
+
+/**
+ * @summary [500] Return a Internal Server Error error object
+ * @function
+ * @param {string} detail
+ * @returns {Object} Internal Server Error error object
+ */
+const internalServerError = detail => new JSONAPIError(error('500', 'Internal Server Error', detail));
```

Comment:
For the handlers for 500, 409, 404, 403, and 401, the parameters are `status`, `title`, `detail`, when they should be `status`, `title`, `code`, `detail`

Reply:
Good catch, fixed.